### PR TITLE
Fix #16 (Responsibilities mixup when resolving with Promise)

### LINKED ISSUE #16: Responsibilities mixup when resolving with Awaitable
> If an Awaitable is resolved with another Awaitable, the Awaitable MUST keep in pending state until the passed Awaitable is resolved. Thus, the value of an Awaitable can never be an Awaitable.

Yes, the value of an Awaitable _MUST NOT_ be an Awaitable.

But an implementation might choose to e.g. not accept resolutions with Awaitables or similar. This is not our responsibility, but the responsibility of the implementing library to have sane handling for that.

Thus I propose this rewording:

> An Awaitable result MUST NOT be an Awaitable.

That's it. Awaitable resolvers can do what they want here. As long as no Awaitable is passed as result.

### PR DESCRIPTION
**Pull request by bwoebi**

### CODE REVIEW

**Review by trowski — APPROVED** [review]

File: META.md [hunk 1]
```diff
@@ -23,3 +23,9 @@ Even though we are not using a thenable, `Promise` is a more recognizable and ac
 ## Creation of Promises
 
 Promise creation and managing is out of scope of this specification, as managing never shall cross library boundaries and thus does not need to be interoperable at all; each library shall resolve the Promise it created itself.
+
+## Results must not be a Promise
+
+The specification recommends whenever the result of a `Promise` is being set to another `Promise`, it should be set to the value the other `Promise` has been or will be resolved with.
```
Comment (kelunik) [inline META.md:29]: _recommends_ doesn't fit here IMO.

File: META.md [hunk 2]
```diff
@@ -23,3 +23,9 @@ Even though we are not using a thenable, `Promise` is a more recognizable and ac
 ## Creation of Promises
```
Comment (kelunik) [inline META.md:23]: I'd just rename this section to `Resolution of Promises` and remove the below header, so combine it into one section.

File: META.md (hunk 1 above)
Comment (kelunik) [reply META.md:29]: The result of a `Promise` can never be set to another `Promise`, as that's not allowed.

**Review by kelunik — CHANGES_REQUESTED** [review]

File: META.md (hunk 1 above)
Comment (bwoebi) [reply META.md:29]: That's true and said in the README. This is the META document, giving a recommendation.

Feel free to push with better wording.

**Comment by bwoebi** [comment]
Closing in favor of much better wording in #24.

**Pull request closed without merging**